pre_commit/main.py

```diff
@@ -122,6 +123,11 @@ def main(argv=None):
     if len(argv) == 0:
         argv = ['run']
     args = parser.parse_args(argv)
+    if hasattr(args, 'files') and args.files:
```

Comment:
`hasattr` will always be `True` here

Add `default=[]` to the definition of `--files` above and you won't need this branch at all :D